### PR TITLE
ct: Add correct file and line to surefire report

### PR DESCRIPTION
**Pull request by garazdawi**
This PR adds file and line number info to the surefire report so that tools (such as [publish-unit-test-result-action](https://github.com/EnricoMi/publish-unit-test-result-action)) can locate where a failed test case is.

The algorithm for finding the source file is very naive, so most users will most likely need to post-process the XML file in order to get the correct path.

### CODE REVIEW

**Review by u3s — APPROVED** [review]
please try to find the test case that is MIA :-)

File: lib/common_test/test/ct_surefire_SUITE_data/surefire_SUITE.erl [hunk 1]
```diff
@@ -76,12 +77,17 @@ break(_Config) ->
     test_server:break(""),
     ok.
 
-tc_ok(_Config) ->
-    ok.
+tc_ok(Config) when is_list(Config) ->
+    ok;
+tc_ok(_) ->
```
Comment (u3s) [inline lib/common_test/test/ct_surefire_SUITE_data/surefire_SUITE.erl:82]: Whats the purpose of this clause?

**Review by u3s — APPROVED** [review]

File: lib/common_test/test/ct_surefire_SUITE_data/surefire_SUITE.erl (hunk 1 above)
Comment (garazdawi) [reply lib/common_test/test/ct_surefire_SUITE_data/surefire_SUITE.erl:82]: To check that the line number reported is the first clause of a function and that multiple clauses actually work.

Comment (u3s) [reply lib/common_test/test/ct_surefire_SUITE_data/surefire_SUITE.erl:82]: Does it also work if you reverse the clause order? So the 2nd matches?

**Comment by max-au** [comment]
I wonder how this change is reflected in the report?

This is how junit_report.xml rendered with GitLab looks before the change:
before

And that's after the change:
after

(I'm interested in improving cth_surefire to a degree that it may be used for test debugging).

File: lib/common_test/test/ct_surefire_SUITE_data/surefire_SUITE.erl (hunk 1 above)
Comment (garazdawi) [reply lib/common_test/test/ct_surefire_SUITE_data/surefire_SUITE.erl:82]: No, it takes the first clause no matter what.

**Comment by garazdawi** [comment]
I had a quick look at the [gitlab source](https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/gitlab/ci/parsers/test/junit.rb#L81-91) and it seems like they don't parse the `line` number, but they do parse the `file`. What they do with it is unclear though. Maybe the file has to exist? So if you make sure to rewrite the paths in the surefire XML file it will point to the correct place?

**Pull request merged**